tests/executors/test_tcp_executor.py

```diff
@@ -5,40 +5,51 @@
 """
 
 import logging
+import socket
 
 import pytest
 from _pytest.logging import LogCaptureFixture
 
 from mirakuru import AlreadyRunning, TCPExecutor, TimeoutExpired
 from tests import HTTP_SERVER_CMD
 
-PORT = 7986
 
+def _find_free_port():
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```

Comment:
@booxter could you explain to me what is happening here, so that socket gets a random (?), unused port returned?

Reply:
Yes, it asks to allocate a socket with any free port. Then it sets an option on the allocated socket to allow it to be reused by another process (REUSEADDR) since we don't actually need this allocated socket object and want to use the port number elsewhere.